src/bin/spin.rs

```diff
@@ -54,12 +52,7 @@ enum SpinApp {
     New(NewCommand),
     Add(AddCommand),
     Up(UpCommand),
-    #[clap(subcommand)]
-    Cloud(CloudCommands),
-    // acts as a cross-level subcommand shortcut -> `spin cloud deploy`
-    Deploy(DeployCommand),
-    // acts as a cross-level subcommand shortcut -> `spin cloud login`
-    Login(LoginCommand),
+    Cloud(CloudCommand),
```

Comment:
In the original `spin cloud` work, I recall people strongly wishing to keep the `spin deploy` and `spin login` aliases.  (Which had to be implemented as commands because clap doesn't allow "cross level" aliases.)

Reply:
@itowlson I updated the implementation to be an "enlightened" plugin experience, wherein the `external` crate is now aware of the `cloud`, `login`, and `deploy` commands and looks to use the `cloud` plugin for any of them.

Reply:
Will that suffice for `login` and `deploy` to appear in help text?  I seem to recall that was asked for when the `cloud` subcommand was originally created, though my memory may be faulty of course!

Reply:
No i dont believe it will. Do we want `cloud` to also appear in the help text?

Reply:
This is a good call out for a section of the SIP (cc @bacongobbler)

Reply:
I believe that was the original plan (and one of the open problems from before).  The sense of the discussion was that this should all feel seamless from a user point of view, as if it was built in.  I don't know, though, whether that remains the feeling, or how strongly it's held!

Reply:
I think it is a good goal. I'll try to iterate on it

Reply:
@itowlson I updated the implementation so that `cloud` `login` and `deploy` are still displayed as CLI subcommands but are executed by the plugin

Reply:
🎉 Perfect - thanks!